lib/vm-watcher.js

```diff
@@ -0,0 +1,341 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2015, Joyent, Inc.
+ */
+
+/*
+ * This module exists to allow consumers to get events when VM changes occur on
+ * a CN. This is updated to work on all currently supported platform versions of
+ * SDC.
+ *
+ * In order to function vm-watcher uses some number of "watcher" submodules.
+ * These watchers implement an interface where they have:
+ *
+ *  * a constructor that takes an 'opts' object that contains at least:
+ *     * a 'log' bunyan logger
+ *     * an 'updateVm' function (vmUuid, updateType, updateObj) where:
```

Comment:
It might clarify things to name `'updateVm'` a callback instead of a function, to convey the meaning that it's used to notify the component that instantiated the watcher that something happened.